autogpt_platform/backend/schema.prisma

```diff
@@ -196,24 +302,65 @@ model AgentGraphExecutionSchedule {
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
-  agentGraphId      String
-  agentGraphVersion Int        @default(1)
-  AgentGraph        AgentGraph @relation(fields: [agentGraphId, agentGraphVersion], references: [id, version], onDelete: Cascade)
+  configuredAgentId String
+  configuredAgent   ConfiguredAgent @relation(fields: [configuredAgentId], references: [id], onDelete: Cascade)
 
   schedule  String // cron expression
   isEnabled Boolean @default(true)
-  inputData String // JSON serialized object
+
+  // Allows triggers to be routed down different execution paths in an agent graph
+  triggerIdentifier String
 
   // default and set the value on each update, lastUpdated field has no time zone.
   lastUpdated DateTime @updatedAt
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId       String
+  user         User    @relation(fields: [userId], references: [id], onDelete: Cascade)
+  Agent        Agent?  @relation(fields: [agentId, agentVersion], references: [id, version])
+  agentId      String?
+  agentVersion Int?
 
   @@index([isEnabled])
 }
 
+enum HttpMethod {
+  GET
+  POST
+  PUT
+  DELETE
+  PATCH
+}
+
+model WebhookTrigger {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  configuredAgentId String
+  ConfiguredAgent   ConfiguredAgent @relation(fields: [configuredAgentId], references: [id])
+
+  method  HttpMethod
+  urlSlug String
+
+  // Allows triggers to be routed down different execution paths in an agent graph
+  triggerIdentifier String
+
+  isActive           Boolean   @default(true)
+  lastReceivedDataAt DateTime?
+  isDeleted          Boolean   @default(false)
+  Agent              Agent?    @relation(fields: [agentId, agentVersion], references: [id, version])
+  agentId            String?
+  agentVersion       Int?
+
+  @@index([configuredAgentId])
+}
```

Comment:
This doesn't match the implementation I'm working on atm so will need updating after that merges.

Reply:
np, will do